include/mbedtls/asn1.h

```diff
@@ -145,9 +145,9 @@ extern "C" {
  */
 typedef struct mbedtls_asn1_buf
 {
-    int MBEDTLS_PRIVATE(tag);                /**< ASN1 type, e.g. MBEDTLS_ASN1_UTF8_STRING. */
-    size_t MBEDTLS_PRIVATE(len);             /**< ASN1 length, in octets. */
-    unsigned char *MBEDTLS_PRIVATE(p);       /**< ASN1 data, e.g. in ASCII. */
+    int tag;                /**< ASN1 type, e.g. MBEDTLS_ASN1_UTF8_STRING. */
```

Comment:
Please document that these fields are read only (even though they are not const)

Reply:
I don't think they're read-only. It's ok to pass a custom `mbedtls_asn1_buf` to some functions that normally work on the result of parsing, but that applications may occasionally want to call with data that isn't the result of parsing.